presidio_evaluator/evaluation/span_evaluator.py

```diff
@@ -0,0 +1,390 @@
+                continue
+            # Skip if token is a stop word
+            if token in self.stop_words:
+                continue
+            normalized.append(token)
+        return normalized
+
+    def merge_adjacent_spans(
+        self, spans: List[Span], df: pd.DataFrame
+    ) -> List[Span]:
+        """
+        Merge adjacent spans of the same entity type.
+        
+        :param spans: List of Span objects to potentially merge
+        :param df: Original dataframe to reference token positions
+        :return: List of merged Span objects
+        """
+        if not spans:
+            return []
+
+        # Sort spans by start position
+        spans = sorted(spans, key=lambda x: x.start_position)
+        merged = []
+        current = spans[0]
+        
+        for next_span in spans[1:]:
+            # Check if spans are adjacent and of same type
+            if (
+                current.entity_type == next_span.entity_type 
+                and self._are_spans_adjacent(current, next_span, df)
+            ):
+                # Get all tokens between start of first span and end of second span
+                tokens = df[
+                    (df["start"] >= current.start_position) & 
+                    (df["start"] < next_span.end_position)
+                ]["token"].tolist()
+                
+                # Merge spans
+                current = Span(
+                    entity_type=current.entity_type,
+                    entity_value=tokens,
+                    start_position=current.start_position,
+                    end_position=next_span.end_position
+                )
+            else:
+                merged.append(current)
+                current = next_span
+        
+        merged.append(current)
+        return merged
+
+    def _are_spans_adjacent(
+        self, span1: Span, span2: Span, df: pd.DataFrame
+    ) -> bool:
+        """
+        Check if two spans are adjacent in the original text.
+        
+        :param span1: First span
+        :param span2: Second span
+        :param df: DataFrame containing token information
+        :return: True if spans are adjacent, False otherwise
+        """
+        # Get tokens between spans' end and start positions
+        between_tokens = df[
+            (df["start"] >= span1.end_position) & 
+            (df["start"] < span2.start_position)
+        ]["token"].tolist()
+        
+        # Check if there are only spaces or punctuation between spans
+        return all(
+            all(c in self.punctuation or c.isspace() for c in token)
+            for token in between_tokens
+        )
+
+    def calculate_iou(self, span1: Span, span2: Span, df: pd.DataFrame) -> float:
+        """
+        Calculate the Intersection over Union (IoU) between two spans.
+        
+        :param span1: First span
+        :param span2: Second span
+        :param df: DataFrame containing token information
```

Comment:
Is this correct? if yes, let's specify
```suggestion
        :param df: DataFrame containing token information for a specific sentence id
```